docs/dual-connection-states.md

```diff
@@ -13,8 +13,24 @@ the replica is ready to serve consistent responses.
 
 ![DualConnection states](dual-connection-states.png "DualConnection states")
 
-If we ended in `ReplicaConnection` state, then the next query will go the same way.
+If we ended in `ReplicaConnection` or `ReadFromMainConnection` state, then the next query will go the same way.
 `MainConnection` is a permanent state and `DualConnection` re-use main connection for
 next queries.
 
+## States
+
+- `NoConnection` - when we create a new `DualConnection` it doesn't allocate
+  any real database connection.
+- `ReplicaConnection` - connection to the replica database has been established. It will be
+  re-used until we need to switch to either `MainConnection` or `ReadFromMainConnection`.
+- `MainConnection` - connection to the main database has been established and used to do
+  writes to the database. `DualConnection` will keep using it until closed.
+  The main reasons not to switch to replica from this state:
+    - to avoid affecting transactions
+    - to avoid affecting locks
+- `ReadFromMainConnection` - connection to the main database has been established, but only
```

Comment:
This one is new. It's missing from the UML

Reply:
> It's missing from the UML

What do you mean? The UML is updated.

Reply:
My, bad, I viewed rich-diff of the `.md`, but I should look at diff of `.puml` and `.png`.